addons/mail/static/src/core/common/message_model.js

```diff
@@ -347,12 +365,16 @@ export class Message extends Record {
             mentionedChannels,
             mentionedPartners,
         });
+        let prettyBody = await prettifyMessageContent(body, validMentions);
+        if (this.thread.model === "discuss.channel") {
+            prettyBody = `<odoo-markdown>${prettyBody}</odoo-markdown>`;
+        }
```

Comment:
`Message.thread` is optional. Right now I don't think you can edit a message that is not linked to thread, but this could happen so better play safe and make code more robust to this possibility

Reply:
That is great, but I don't see an alternative to this test on discuss.channel?

Reply:
Just ignore if the thread is not defined?